### PR TITLE
Enhance look_up_table op 

### PR DESCRIPTION
**Pull request by chengduoZH**
fix https://github.com/PaddlePaddle/Paddle/issues/8933

### CODE REVIEW

File: paddle/fluid/operators/lookup_table_op.cc [hunk 1]
```diff
@@ -85,6 +91,44 @@ or not. And the output only shares the LoD information with input Ids.
   }
 };
 
+class ConcatRowsOpMaker : public framework::OpProtoAndCheckerMaker {
```
Comment (jacquesqiao) [inline paddle/fluid/operators/lookup_table_op.cc]: I think this name is not so good and maybe we just need a python layer but not a new operator

Comment (chengduoZH) [reply paddle/fluid/operators/lookup_table_op.cc]: Thanks for your review @jacquesqiao 
The parameters' meaning of ConcatRowsOp are different with LookUpTableOp, and the two operations' function is also different, so I define the new operation(ConcatRows). And I have added some annotation in the code.

Is your meaning that ConcatRowsOp is unnecessary?

File: python/paddle/fluid/tests/unittests/test_lookup_table_op.py [hunk 2]
```diff
@@ -47,5 +49,53 @@ def test_check_grad(self):
...
+        ids_tensor.set(np_array, place)
+
+        # create and initialize W Variable
+        W = scope.var('W').get_tensor()
+        W_array = np.full((height, row_numel), 1.0).astype("float32")
+        for i in range(height):
+            W_array[i] *= i
+        W.set(W_array, place)
+
+        Out = scope.var('Out').get_selected_rows()
```
Comment (jacquesqiao) [inline python/paddle/fluid/tests/unittests/test_lookup_table_op.py]: remote these lines

File: python/paddle/fluid/tests/unittests/test_lookup_table_op.py [hunk 3]
```diff
@@ -47,5 +49,53 @@ def test_check_grad(self):
...
+
+        # create and run concat_rows_op operator
+        concat_rows_op = Operator("lookup_table", W='W', Ids='Ids', Out='Out')
+        concat_rows_op.run(scope, place)
+
+        # get and compare result
+        result_array = np.array(Out_tensor)
+
+        for idx, row in enumerate(rows):
+            assert (row == result_array[idx]).all()
```
Comment (jacquesqiao) [inline python/paddle/fluid/tests/unittests/test_lookup_table_op.py]: add comment about this usage of numpy.array

Comment (chengduoZH) [reply python/paddle/fluid/tests/unittests/test_lookup_table_op.py]: Done

File: python/paddle/fluid/tests/unittests/test_lookup_table_op.py (hunk 2 above)
Comment (chengduoZH) [reply python/paddle/fluid/tests/unittests/test_lookup_table_op.py]: Done

**Pull request merged**